Library/Formula/mongoose.rb

```diff
@@ -1,16 +1,16 @@
-require 'formula'
-
 class Mongoose < Formula
-  homepage 'https://github.com/valenok/mongoose'
-  url 'https://github.com/valenok/mongoose/archive/5.1.tar.gz'
-  sha1 '0a2d0b83a7f5650a6e6c50fd7ba18a06ebe79335'
+  homepage "https://github.com/cesanta/mongoose"
+  url "https://github.com/cesanta/mongoose/archive/5.6.tar.gz"
+  sha256 "cc2557c7cf9f15e1e691f285a4c6c705cc7e56cb70c64cb49703a428a0677065"
 
   def install
-    cd 'build' do
-      system 'make all'
-      bin.install "mongoose"
-    end
-    include.install 'mongoose.h'
-    prefix.install 'examples', 'docs'
+    system ENV.cc, "-dynamiclib", "mongoose.c", "-o", "libmongoose.dylib"
+    include.install "mongoose.h"
+    lib.install "libmongoose.dylib"
+    share.install "examples", "docs", "jni"
+  end
+
+  test do
+    system ENV.cc, "#{share}/examples/hello_world/hello_world.c", "-o", "hello_world", "-lmongoose"
```

Comment:
I tried your suggestion. It worked, but there are several problems:
* The firewall popup a message box to ask for permission
* The program will listen on port 8080, which may be conflict with other problem

So I'm wondering whether or not it's appropriate to add such a test?